client/core/state_transition.go

```diff
@@ -158,27 +165,43 @@ func (st *StateTransition) preCheck() error {
 			return ErrNonceTooLow
 		}
 	}
+
+	// prepay stability fee if it applies
+	if st.evm.StabilizationLevel != 0 {
+		computeFee := new(big.Int).Mul(new(big.Int).SetUint64(st.msg.Gas()), st.gasPrice)
+		if stabilityFee := stability.CalcFee(computeFee, st.evm.StabilizationLevel, st.msg.Value()); stabilityFee.Cmp(common.Big0) > 0 {
+			if st.state.GetBalance(st.msg.From()).Cmp(stabilityFee) < 0 {
+				return errInsufficientBalanceForStabilityFee
+			}
+			st.state.SubBalance(st.msg.From(), stabilityFee)
+			st.initialStabilityFee = stabilityFee
+		}
+	}
+
 	return st.buyGas()
 }
 
 // TransitionDb will transition the state by applying the current message and
 // returning the result including the the used gas. It returns an error if it
 // failed. An error indicates a consensus issue.
-func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bool, err error) {
-	if err = st.preCheck(); err != nil {
-		return
-	}
+func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, stabilityFee *big.Int, failed bool, err error) {
```

Comment:
this return signature is getting too long, want to consider represent it with a struct?